### DISCUSSION TITLE
memory consumtion control

### DISCUSSION THREAD

**Discussion by FROL256** [mitsuba-renderer/mitsuba3#202] [Q&A]
Hi! I wonder whether it is possible to reduce memory consumption of Mitsuba3 in any way like doing several "tiles" or "batches" per render or something like that.

I'm a bit confused that 128x128 with 8 spp takes about 15Gb on my machine with a quite simple scene (Mitsuba Material Ball scene).   

And when getting more spp, memory consumption grows almost lineary. Probably i don't understand things about differential rendering in the right way, but ... i would expect that we can process separate samples or tiles of the image and get several unbiased estimates of gradient to later just add them together. In this way i agree to wait more for more samples but would expect some constant memory consumption. 

Thank you very much!

## My system configuration
* Platform: Linux Mint 20.3 (Cinnamon), i9-10940X CPU @ 3.30GHz, 46.8 GiB Mem.
* Python version: 3.8
* Mitsuba 3 version: Successfully installed drjit-0.2.1 mitsuba-3.0.1
* Compiled variants: 'llvm_ad_rgb'

**Comment by FROL256** [comment]
Btw, i have tried both rb (Radiative Backpropagation) and prb (Path Replay Backpropagation) integrators and it still is quite memory hungry )

**Reply by Speierers** [reply]
Actually that shouldn't be the case. Could you share a simplified python script to reproduce this?

**Reply by FROL256** [reply]
Sure, in this issue https://github.com/mitsuba-renderer/mitsuba3/issues/201

**Comment by FROL256** [comment]
I have orerride integrator and resolution inside XML files and that's done. Thank you very much! Now it is not so hungry )

```
   <default name="spp" value="256"/>
    <default name="width" value="256"/>
    <default name="height" value="256"/>
    <default name="max_depth" value="-1"/>

    <integrator type="prb">
        <integer name="max_depth" value="$max_depth"/>
    </integrator>
```
